okhttp/src/commonMain/kotlin/okhttp3/internal/-ResponseBodyCommon.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2022 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package okhttp3.internal
+
+import okhttp3.MediaType
+import okhttp3.ResponseBody
+import okhttp3.ResponseBody.Companion.asResponseBody
+import okio.Buffer
+import okio.BufferedSource
+import okio.ByteString
+import okio.IOException
+import okio.use
+
+internal fun ResponseBody.commonBytes() = commonConsumeSource(BufferedSource::readByteArray) { it.size }
+
+internal fun ResponseBody.commonByteString() = commonConsumeSource(BufferedSource::readByteString) { it.size }
+
+internal inline fun <T : Any> ResponseBody.commonConsumeSource(
```

Comment:
Take it or leave it: is the amount of code saved here worth the abstraction? The original code was really simple, this is pretty fancy.

Reply:
TIL: this abstraction predates this refactor. I had no idea. I don’t care for it!

Reply:
FYI @JakeWharton !

Reply:
I'll follow up on this, there is a fair amount of logic there.